### PR TITLE
Add PublishPrebuiltReportData target to make rolling build prebuilt data available

### PR DESCRIPTION
**Pull request by dagood**
Publishes the prebuilt report data for tarball builds to a blob storage location. Just needs the `publish.blobStorage.*` variables specified in the internal build def.

https://github.com/dotnet/source-build/issues/851

### CODE REVIEW

**Comment by crummel** [comment]
Think there's value in keeping historical versions around as well as latest?  @dseefeld and I were looking back at previous reports when we were looking at #867.

**Comment by dagood** [comment]
Hmm. They are kept around in the AzDO build artifacts, and they're reproducible for all practical purposes. However, the builds aren't retained for very long, and in theory, feeds can be mutated, so it wouldn't hurt to save each one to blob storage with a build-versioned path as well. I'll add that.

File: .vsts.pipelines/jobs/ci-linux.yml [hunk 1]
```diff
@@ -195,6 +195,15 @@ jobs:
...
       ArtifactType: Container
 
+  # Publish prebuilt report data to blob storage.
+  - template: ../steps/publish-prebuilt-data-sh.yml
+    parameters:
+      relativeBlobPath: prebuilt-data/$(Build.SourceBranchName)/latest/$(artifactName)
+  # Also publish per-build data to a unique location for future use.
+  - template: ../steps/publish-prebuilt-data-sh.yml
+    parameters:
+      relativeBlobPath: prebuilt-data/$(Build.SourceBranchName)/rolling/$(Build.SourceVersion)/$(Build.DefinitionName)/$(Build.BuildNumber)/$(artifactName)
```
Comment (dagood) [inline .vsts.pipelines/jobs/ci-linux.yml]: I figure I have to include at least `Build.SourceVersion`, `Build.DefinitionName`, and `Build.BuildNumber` in here to make sure each build has a unique path, and make it reasonable to know which commit the data is for, I'm not sure about the order though. @dseefeld @crummel Does what I have make sense to you?

Comment (crummel) [reply .vsts.pipelines/jobs/ci-linux.yml]: I think that makes sense.  I could see swapping `DefinitionName` and `BuildNumber` too but I'm fine with this, I can't imagine we'll have too many builds for the same commit.

Comment (dagood) [reply .vsts.pipelines/jobs/ci-linux.yml]: Hmm, yeah, agreed the swap makes more sense. Thanks.

**Pull request merged**